### ISSUE TITLE
CA1062 False positive when using ??= operator

### ISSUE THREAD

**Issue by BL-SB** [dotnet/roslyn-analyzers#3634]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v3.0.0 (Latest)

#### Diagnostic ID

Example: [CA1716](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1062)

#### Repro steps

```
public void ParameterTest(Dictionary<string,string> dict = null)
{
    s ??= new Dictionary<string, string>();
    SetParameter(dict);
}
```
#### Expected behavior

Build is successful because parameter will be an empty Dictionary before calling SetParameter.

#### Actual behavior

In call to SetParameter dict is marked with CA1062

**Comment by Evangelink** [comment]
Assuming the code snippet you wanted to provide is

```csharp
public void ParameterTest(Dictionary<string, string> dict = null)
{
    dict ??= new Dictionary<string, string>();
    SetParameter(dict);
}
```

I cannot reproduce the issue with the latest master code. Here are the full tests I have used:

```csharp
[Fact, WorkItem(3634, "https://github.com/dotnet/roslyn-analyzers/issues/3634")]
public async Task NullConditionalAssignmentOperator_NullableEnableContext_NoDiagnostic()
{
	await new VerifyCS.Test
	{
		TestCode = @"
#nullable enable
using System.Collections.Generic;

public class C
{
    public void ParameterTest(Dictionary<string, string>? dict = null)
    {
        dict ??= new Dictionary<string, string>();
        SetParameter(dict);
    }

    private void SetParameter(Dictionary<string, string> dict)
    {
    }
}",
		LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp8
	}.RunAsync();
}

[Fact, WorkItem(3634, "https://github.com/dotnet/roslyn-analyzers/issues/3634")]
public async Task NullConditionalAssignmentOperator_NonNullableEnableContext_NoDiagnostic()
{
	await new VerifyCS.Test
	{
		TestCode = @"
using System.Collections.Generic;

public class C
{
    public void ParameterTest(Dictionary<string, string> dict = null)
    {
        dict ??= new Dictionary<string, string>();
        SetParameter(dict);
    }

    private void SetParameter(Dictionary<string, string> dict)
    {
    }
}",
		LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp8
	}.RunAsync();
}
```

@mavasani Do you want me to create a PR with the 2 tests? Also, I think the ticket could be closed.

**Comment by genlu** [comment]
@Evangelink Yes, please add those tests and close this issue. Thanks!
